src/graphic/Fast3D/Fast3dWindow.h

```diff
@@ -29,6 +29,8 @@ class Fast3dWindow : public Ship::Window {
     bool IsFullscreen() override;
     bool IsRunning() override;
     const char* GetKeyName(int32_t scancode) override;
+    bool ShouldForceCursorVisibility();
+    void SetForceCursorVisibility(bool visible);
```

Comment:
why is the cursor visibility stuff in f3d window instead of base window?

Reply:
I think only because I didn't realize that some non-virtual functions still existed there.